tapchannel/aux_traffic_shaper.go

```diff
@@ -277,50 +277,72 @@ func paymentBandwidthAssetUnits(htlcAssetAmount, computedLocal uint64,
 }
 
 // paymentBandwidth returns the available payment bandwidth of the channel based
-// on the asset rate of the RFQ quote that is included in the HTLC and the asset
-// units of the local balance.
+// on the list of availalbe RFQ IDs. If any of those IDs matches the channel, we
+// calculate the bandwidth based on its asset rate.
 func (s *AuxTrafficShaper) paymentBandwidth(htlc *rfqmsg.Htlc,
 	localBalance uint64, linkBandwidth, minHtlcAmt lnwire.MilliSatoshi,
 	fundingChan *cmsg.OpenChannel) (lnwire.MilliSatoshi, error) {
 
-	// If the HTLC doesn't have an RFQ ID, it's incomplete, and we cannot
-	// determine the bandwidth.
-	if htlc.RfqID.ValOpt().IsNone() {
-		log.Tracef("No RFQ ID in HTLC, cannot determine matching " +
-			"outgoing channel")
+	// If the HTLC doesn't have any available RFQ IDs, it's incomplete, and
+	// we cannot determine the bandwidth.
+	if htlc.AvailableRfqIDs.IsNone() {
+		log.Tracef("No available RFQ IDs in HTLC, cannot determine " +
+			"matching outgoing channel")
 		return 0, nil
 	}
 
-	// For every other use case (i.e. a normal payment with a negotiated
-	// quote or a multi-hop keysend that also uses a quote), we need to look
-	// up the accepted quote and determine the outgoing bandwidth in
-	// satoshis based on the local asset balance.
-	rfqID := htlc.RfqID.ValOpt().UnsafeFromSome()
+	// Retrieve the available RFQ IDs.
+	avaliableIDs := htlc.AvailableRfqIDs.UnsafeFromSome().Val.IDs
+
+	// For this rfqID we'll fetch the corresponding quote and rate.
 	acceptedSellQuotes := s.cfg.RfqManager.PeerAcceptedSellQuotes()
 	acceptedBuyQuotes := s.cfg.RfqManager.LocalAcceptedBuyQuotes()
 
-	sellQuote, isSellQuote := acceptedSellQuotes[rfqID.Scid()]
-	buyQuote, isBuyQuote := acceptedBuyQuotes[rfqID.Scid()]
+	for _, rfqID := range avaliableIDs {
+		sellQuote, isSellQuote := acceptedSellQuotes[rfqID.Scid()]
+		buyQuote, isBuyQuote := acceptedBuyQuotes[rfqID.Scid()]
 
-	var (
-		rate      rfqmsg.AssetRate
-		specifier asset.Specifier
-	)
-	switch {
-	case isSellQuote:
-		rate = sellQuote.AssetRate
-		specifier = sellQuote.Request.AssetSpecifier
+		var (
+			rate      rfqmsg.AssetRate
+			specifier asset.Specifier
+		)
+		switch {
+		case isSellQuote:
+			rate = sellQuote.AssetRate
+			specifier = sellQuote.Request.AssetSpecifier
+
+		case isBuyQuote:
+			rate = buyQuote.AssetRate
+			specifier = buyQuote.Request.AssetSpecifier
+
+		default:
+			return 0, fmt.Errorf("no accepted quote found for RFQ "+
+				"ID %x (SCID %d)", rfqID[:], rfqID.Scid())
+		}
 
-	case isBuyQuote:
-		rate = buyQuote.AssetRate
-		specifier = buyQuote.Request.AssetSpecifier
+		bandwidth, err := s.paymentBandwidthRFQ(
+			rfqID, rate, specifier, localBalance, linkBandwidth,
+			minHtlcAmt, fundingChan,
+		)
+		if err != nil {
+			return 0, err
+		}
 
-	default:
-		return 0, fmt.Errorf("no accepted quote found for RFQ ID "+
-			"%x (SCID %d)", rfqID[:], rfqID.Scid())
+		if bandwidth > 0 {
```

Comment:
So we explicitly just return the first bandwidth we find? To kind of almost randomize which channel we use first? Might be worth a comment here or in the method Godoc about _why_ we're taking the first vs. the largest for example.

Reply:
I think "Given we establish 1 quote per peer, only one of the RFQ IDs in the array should produce some bandwidth." would probably be a good comment to add above this statement.

Reply:
Comment still missing.